server/task/task-api/src/test/java/org/apache/james/task/TaskManagerContract.java

```diff
@@ -213,6 +213,68 @@ default void getStatusShouldReturnCompletedWhenRunSuccessfully() {
             .isEqualTo(TaskManager.Status.COMPLETED);
```

Comment:
I'm missing a test that asserts that the task manager updates additionalInformation during the execution of a task.

I believe this can be achived via the use of a latch.

(If I understood it well, your work here do not allow that yet, but having the test describing the limitations of your work will be valuable)

Reply:
This test will come in the next PR. 
I have just pushed an update adding tests for the `failed` and `canceled` and terminaison event.

Reply:
(I just find it sad that, as off today this limitation is not documented in the code)

Reply:
Today it's not a limitation, the code is not binded.
This raises an interesting question: when binding the code, do we merge all the code on a branch while the core functionalities are not finished, or do we take the risk to have a master with a limited set of functionalities during some time?

Reply:
Having a 'product' with a limited set of feature supported is OK to me as long as we get the tests disabled, and a ticket with that.

I'm in favour of binding early thus don't like the feature branch approach much.

Reply:
I'm completely against the feature branch concept.
We may need transient products to start having the right testing in place then promote the feature into a supported product once it's ready.